Kerberos.NET/Entities/KerberosConstants.cs

```diff
@@ -46,25 +60,14 @@ internal static class KerberosConstants
             return null;
         }
 
-        internal static Guid GetRequestActivityId()
-        {
-            var counter = Interlocked.Increment(ref requestCounter);
-
-            var b = BitConverter.GetBytes(counter);
-
-            return new Guid(Pid, 0, 0, b[7], b[6], b[5], b[4], b[3], b[2], b[1], b[0]);
-        }
+        internal static Guid GetRequestActivityId() => Guid.NewGuid();
 
         internal static int GetNonce()
         {
-            // .NET Runtime guarantees operations on variables up to the natural
-            // processor pointer size are intrinsically atomic, but there's no
-            // guarantee we'll be running in a 64 bit process on a 64 bit processor
-            // so maybe let's not give the system an opportunity to corrupt this
-
-            var counter = Interlocked.Increment(ref nonceCounter);
+            var bytes = new byte[4];
+            Rng.GetBytes(bytes);
```

Comment:
`RandomNumberGenerator` has a span-overload, so this allocation could be avoided.

```suggestion
            Span<byte> bytes = stackalloc byte[4];
            Rng.GetBytes(bytes);
```

---

Side note: if this would be really high-perf, so to eke out more perf instead of stackalloc this could be used:
```c#
int tmp;
var bytes = new Span<byte>(&tmp, sizeof(int));
```
(it requires `unsafe`).
This has smaller codegen that the stackalloc, as no stack-guards need to be emitted by the JIT.

Reply:
The span overload doesn't exist in netstandard2.0 unfortunately, otherwise that's exactly what I'd do.